pkg/pillar/cmd/volumemgr/handlecontent.go

```diff
@@ -11,33 +11,81 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-func handleContentTreeCreate(ctxArg interface{}, key string,
+func handleContentTreeCreateAppImg(ctxArg interface{}, key string,
 	configArg interface{}) {
 
-	log.Infof("handleContentTreeCreate(%s)", key)
+	log.Infof("handleContentTreeCreateAppImg(%s)", key)
 	config := configArg.(types.ContentTreeConfig)
 	ctx := ctxArg.(*volumemgrContext)
-	updateContentTree(ctx, config)
-	log.Infof("handleContentTreeCreate(%s) Done", key)
+	status := createContentTreeStatus(ctx, config, types.AppImgObj)
+	updateContentTree(ctx, status)
+	log.Infof("handleContentTreeCreateAppImg(%s) Done", key)
 }
 
-func handleContentTreeModify(ctxArg interface{}, key string,
+func handleContentTreeCreateBaseOs(ctxArg interface{}, key string,
+	configArg interface{}) {
+
+	log.Infof("handleContentTreeCreateBaseOs(%s)", key)
+	config := configArg.(types.ContentTreeConfig)
+	ctx := ctxArg.(*volumemgrContext)
+	status := createContentTreeStatus(ctx, config, types.BaseOsObj)
+	updateContentTree(ctx, status)
+	log.Infof("handleContentTreeCreateBaseOs(%s) Done", key)
+}
```

Comment:
Minor nit: Move baseos create function along with baseos modify function.

Reply:
Since I removed ObjType from ContentTreeConfig I need a way to look up the Status.
Either that means separate functions which embed the ObjType, or looking in both Publications.

Reply:
Ah you meant in the order in the file?

Reply:
Yes